lib/location/Kconfig

```diff
@@ -0,0 +1,147 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig LOCATION
+	bool "Location"
+	depends on NRF_MODEM_LIB
+	help
+	  TODO
+
+if LOCATION
+
+config LOCATION_METHOD_GNSS
+	bool "Allow GNSS to be used for obtaining the location"
+	default y
+
+config LOCATION_METHOD_CELLULAR
+	bool "Allow cellular positioning to be used for obtaining the location"
+	select MULTICELL_LOCATION
+	default y
+
+config LOCATION_METHOD_WIFI
+	bool "Allow WiFi positioning to be used for obtaining the location"
+	depends on WIFI
+	select REST_CLIENT
+	select CJSON_LIB
+	select NET_MGMT
+	select NET_MGMT_EVENT
+	select NET_MGMT_EVENT_INFO
+	select NET_L2_WIFI_MGMT
+	default y if WIFI
+
+if LOCATION_METHOD_GNSS
+config LOCATION_METHOD_GNSS_AGPS_EXTERNAL
+	bool "Requesting A-GPS data is handled by the application outside of Location library."
+	depends on NRF_CLOUD_AGPS
+	default n
+	help
+	  Allow application to download the A-GPS data and merely feed it to Location library for
+	  processing. If enabled, Location library throws event LOC_EVT_GNSS_ASSISTANCE_REQUEST when
+	  assistance is needed.
```

Comment:
If we mention the event, should we also mention the function that should be called.

Reply:
Good question. I would say this is not the right place to document every aspect. But would you rather not mention the event either then?

Reply:
At least we should mention the event in function description and vice versa.
Here we should mention either both or neither one.
While generally mentioning such details in Kconfig is not something I would do, in this case I'm not 100% sure removing them is correct either. Mainly because the functionality is so specific and requires application to do very exact steps.
Maybe I would remove them here and need to double check that the documentation mentions them (outside of function and event descriptions).